src/material-experimental/mdc-chips/chip.ts

```diff
@@ -452,13 +452,21 @@ export class MatChip
 
   private _handleTransitionend = (event: TransitionEvent) => {
     if (event.target === this._elementRef.nativeElement) {
-      this._ngZone.run(() => this._chipFoundation.handleTransitionEnd());
+      // The `handleTransitionEnd` calls `adapter.emitEvent` which we handle by calling
+      // `emitCustomEvent`. We don't need to re-enter the Angular zone. First, there might be no
+      // event listeners for the event we dispatch; second, calling `element.dispatchEvent`
+      // doesn't require Angular to run `tick()`.
+      this._chipFoundation.handleTransitionEnd();
```

Comment:
There's usually a reason for these `ngZone.run` calls (e.g. some issue internally) and as mentioned in another PR, calling `ngZone.run` when the code is already in the zone doesn't make much of a difference.

Reply:
This isn't in the zone since this is called within event listeners which are added outside of the zone.

Reply:
The point of the `ngZone.run` call is to bring them back into the zone before they get passed along to MDC. The listeners are bound outside the zone, because we only care about events on the host node, whereas `transitionend` and `animationend` are going to fire for events from child elements as well.

Reply:
Yeah, I understand what it's doing. I mean do we need to bring it back into the zone for calling `emitCustomEvent`?

Reply:
I think that we do, because the event is bound outside the zone and then we don't actually know what'll happen inside the chip foundation methods.